packages/suite-desktop/src-electron/electron.ts

```diff
@@ -193,6 +196,76 @@ const init = async () => {
 
     mainWindow.loadURL(src);
 
+    // TOR
+    const torSettings = store.getTorSettings();
+    const sess = mainWindow.webContents.session;
+    const toggleTor = async (start: boolean) => {
+        if (start) {
+            if (torSettings.running) {
+                await tor.start();
+            } else {
+                await tor.start();
+            }
+        } else {
+            await tor.stop();
+        }
+
+        torSettings.running = start;
+        store.setTorSettings(torSettings);
+
+        mainWindow.webContents.send('tor/status', start);
+        sess.setProxy({
+            proxyRules: start ? 'socks5://localhost:9050' : '',
+        });
+    };
+
+    if (launchWithTor || torSettings.running) {
+        await toggleTor(true);
+    }
+
+    ipcMain.on('tor/toggle', async (_, start: boolean) => {
+        await toggleTor(start);
+    });
+
+    ipcMain.on('tor/get-address', () => {
+        mainWindow.webContents.send('tor/set-address', torSettings.address);
+    });
+
+    ipcMain.on('tor/set-address', () => async (_: IpcMainEvent, address: string) => {
+        if (torSettings.address !== address) {
+            torSettings.address = address;
+            store.setTorSettings(torSettings);
+
+            if (torSettings.running) {
+                await toggleTor(true);
+            }
+        }
+    });
+
+    session.defaultSession.webRequest.onBeforeRequest({ urls: ['*://*/*'] }, (details, cb) => {
+        const { hostname, protocol } = new URL(details.url);
+
+        // Cancel requests that aren't allowed
+        if (config.allowedDomains.find(d => hostname.endsWith(d)) === undefined) {
+            console.warn(`[Warning] Domain '${hostname}' was blocked.`);
```

Comment:
doesn't it brake the logic of `handleExternalLink` ?

Reply:
It doesn't break the opening of links, but the request still gets intercepted here. That's whats intended, only to allow webView requests to authorized domains.

Reply:
so maybe user should be informed about that? (same as in handleExternalLink case)

Reply:
Now filtering xhr requests only (as it was catching everything). It's showing a warning for domains that are blocked and won't prompt the same warning again for the time of the session.

1e0f157